group/assignment5/main.cpp

```diff
@@ -0,0 +1,96 @@
+#include "AlphabetFromDictionary.hpp"
+#include "codeu_test_lib.h"
+
+int main() {
+    vector<string> dictionary;
+    vector<char> alphabet;
+    bool result;
+    unordered_map<char, int> idxs;
+    vector<bool> expectedChars;
+    vector<bool> actualChars;
+
+    // test 1 - assignment example
+    dictionary = {"ART", "RAT", "CAT", "CAR"};
+    AlphabetFromDictionary afd(dictionary);
+    idxs.clear();
+    expectedChars = vector<bool>(pow(2, CHAR_BIT), false);
+    actualChars = vector<bool>(pow(2, CHAR_BIT), false);
+    result = afd.findAlphabet(alphabet);
+    EXPECT_TRUE(result);
+    EXPECT_EQ(4, alphabet.size());
+    for (char c : {'A', 'R', 'C', 'T'}) {
+        expectedChars[c] = true;
+    }
+    for (int i = 0; i < alphabet.size(); i++) {
+        idxs[alphabet[i]] = i;
+        actualChars[alphabet[i]] = true;
+    }
+    EXPECT_EQ(expectedChars, actualChars);
+    EXPECT_TRUE(idxs['A'] < idxs['R']);
```

Comment:
I think it's ok to have hard-coded things in a specific test. Do you have an idea for a more generic way to check if the order of the characters is correct?
(I don't think we should compare 2 arrays because the order of the characters isn't guaranteed since some characters don't have restrictions on their position in the alphabet, so we can't assume a certain order. Also, creating an array of the expected alphabet is hard-coded too.)